dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,682 @@
+                                        "response status.");
+                                response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+                            }
+
+                            return Result.SKIP_NO_CHAIN;
+                        }
+                    }
+
+                    // Starting the logout
+                    // if it is logout
+                    Logger.debug(this, ()-> "----------------------------- doFilter --------------------------------");
+                    Logger.debug(this, ()-> "- isLogoutNeed = " + isLogoutNeed);
+                    Logger.debug(this, ()-> "- httpServletRequest.getRequestURI() = " + request.getRequestURI());
+
+                    if (isLogoutNeed && session != null && this.isLogoutRequest(request.getRequestURI(),
+                            this.getLogoutPathArray(identityProviderConfiguration))) {
+
+                        if (this.doLogout(response, request, session, identityProviderConfiguration)) {
+
+                            return Result.SKIP_NO_CHAIN;
+                        }
+                    }
+
+                } else {
+
+                    Logger.info(this, "No idpConfig for site '" + request.getServerName()
+                            + "'. No SAML filtering for this request: " + request.getRequestURI());
+                }
+            }
+        } catch (final Exception exception) { // todo: better error handling?
+
+            Logger.debug(this, ()-> "Error [" + exception.getMessage() + "] Unable to get idpConfig for Site '" +
+                    request.getServerName() + "'. Incoming URL: " + request.getRequestURL());
+            response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        }
+
+        return result;
+    } // intercept.
+
+    protected boolean isAnySamlConfigurated() {
+
+        return
+                Try.of(()->this.appsAPI.getAppDescriptor(APPS_SAML_CONFIG_KEY,
+                        APILocator.systemUser())).getOrElseGet(e->Optional.empty()).isPresent();
+    }
+
+
+    protected AutoLoginResult autoLogin(final HttpServletRequest request,
+                                        final HttpServletResponse response,
+                                        HttpSession session,
+                                        final IdentityProviderConfiguration identityProviderConfiguration) {
+
+        final User user = this.getUser(request, identityProviderConfiguration);
+        boolean continueFilter   = true; // by default continue with the filter
+        HttpSession renewSession = session;
+
+        if (null != user) {
+            // we are going to do the autologin, so if the session is null,
+            // create it!
+            try {
+
+                Logger.debug(this, "User with ID '" + user.getUserId() + "' has been returned by SAML Service. User " +
+                        "Map: " + user.toMap());
+            } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)